exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/writer/TestParquetWriter.java

```diff
@@ -859,5 +865,40 @@ public void testLastPageOneNull() throws Exception {
         "cp.`parquet/last_page_one_null.parquet`");
   }
 
+  @Test
```

Comment:
Should there be tests for other code paths introduced in this commit?

* Failures in the parallel tasks (see comments above about handling such failures.)
* Verifying that the various "determine size parallel" code paths are exercised?
* Verify that pages were, indeed, read in parallel?
* What happens if more parallel readers are needed than threads available in the pool?
* What happens with parallelization if the file has 1000 columns (which, presumably, could cause 1K parallel reads?) Some upper limit on parallelization?
* Verify the five decompression paths?

Might the above code need a bit of instrumentation so the test can validate that the parallel reader was, in fact, used and exercised the interesting code paths?